### PR TITLE
Completely refactor repository

### PR DESCRIPTION
**Pull request by CameronDiver**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 node_modules
+build
+docs
```
Comment (hedss) [inline .gitignore]: NL

File: test/mocha.opts [hunk 2]
```diff
@@ -0,0 +1 @@
+--timeout 20000
```
Comment (hedss) [inline test/mocha.opts]: NL

File: package.json [hunk 3]
```diff
@@ -3,14 +3,16 @@
   "version": "0.8.0",
   "description": "Build a docker composition",
   "main": "build/index.js",
-  "types": "lib/index.ts",
+  "types": "build/index.d.ts",
   "scripts": {
+    "clean": "rm -rf build",
```
Comment (dfunckt) [inline package.json]: We usually use `rimraf` which is cross-platform.

File: tsconfig.publish.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+	"extends": "./tsconfig.json",
```
Comment (dfunckt) [inline tsconfig.publish.json]: TIL.

File: package.json (hunk 3 above)
Comment (CameronDiver) [reply package.json]: Ah, that's nice. I'll add that in in a upcoming PR - thanks!

File: tsconfig.publish.json (hunk 4 above)
Comment (CameronDiver) [reply tsconfig.publish.json]: Yeah it means my testing builds don't get plagued with things like unused locals etc when I'm just trying to develop. It's pretty nice :)

**Pull request merged**